velox/functions/sparksql/SplitFunctions.cpp

```diff
@@ -134,10 +135,129 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
               .build()};
 }
 
+class StringToMap final : public exec::VectorFunction {
+ public:
+  StringToMap(char entryDelim, char keyValueDelim)
+      : entryDelim_(entryDelim), keyValueDelim_(keyValueDelim) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    DecodedVector* inputString = decodedArgs.at(0);
```

Comment:
Any fast path from constant vector and flat vector?

Reply:
Hi @mbasmanova, do I need to add fast path optimization for constant/flat vector here? Thanks!

Reply:
The best way to answer this question is to implement the fast paths and benchmark these against generic path. See velox/benchmarks/ExpressionBenchmarkBuilder.h for an easy way to write benchmarks for expressions.